cmd/print.go

```diff
@@ -171,6 +183,33 @@ func (pr *PrintRunner) outputResult(gatewayResources []i2gw.GatewayResources) {
 		}
 		fmt.Println(msg)
 	}
+
+	return resourceCount
+}
+
+func (pr *PrintRunner) output2File(gatewayResources []i2gw.GatewayResources) {
+	file, openFileErr := os.OpenFile(pr.outputFile, os.O_TRUNC|os.O_CREATE|os.O_WRONLY, 0666)
+	if openFileErr != nil {
+		fmt.Println("Error opening file:", openFileErr)
+		return
+	}
+	defer func(file *os.File) {
+		err := file.Close()
+		if err != nil {
+			fmt.Println("Error closing file:", err)
+			return
+		}
+	}(file)
+
+	resourceCount := pr.outputResult(gatewayResources, file)
+
+	if resourceCount == 0 {
+		err := os.Remove(pr.outputFile)
+		if err != nil {
+			fmt.Printf("Error removing file %s: %v\n", pr.outputFile, err)
+		}
+	}
```

Comment:
Creating a file upfront and deleting it if no resources are displayed is a bit dirty in my opinion. I'd slightly refactor the `outputResult` function to extract the count of objects into a dedicated function. Before calling `outputResult` we can check if there are resources. If not, we do not need to create any file and we can just move on.

Reply:
ok, got it.